core/http/client/src/test/java/org/eclipse/rdf4j/http/client/BackgroundGraphResultHangTest.java

```diff
@@ -63,11 +63,9 @@ public void testBGRHang()
 				new ByteArrayInputStream(data.getBytes(Charset.forName("UTF-8"))), Charset.forName("UTF-8"),
 				"http://base.org");
 
+		thrown.expect(QueryEvaluationException.class);
```

Comment:
I am now a little suspicious of this change and its side effects in terms of which thread sees the exception now. I may reverse some of the changes around this if I can prove that it is breaking something, but it may actually be useful behaviour rather than waiting until .hasNext is called to throw the wrapped parse exception.